src/liballoc/string.rs

```diff
@@ -2225,6 +2225,14 @@ impl From<&str> for String {
     }
 }
 
+#[stable(feature = "???", since = "1.43.0")]
+impl From<&mut str> for String {
+    #[inline]
```

Comment:
Maybe add documentation for this? Please read https://github.com/rust-lang/rust/issues/51430#issuecomment-398322434 to know what to document.